influxql/statement_rewriter.go

```diff
@@ -457,3 +454,18 @@ func rewriteSourcesCondition(sources Sources, cond Expr) Expr {
 	}
 	return scond
 }
+
+func rewriteSources2(sources Sources, database string) Sources {
+	if len(sources) == 0 {
+		sources = Sources{&Measurement{Regex: &RegexLiteral{Val: regexp.MustCompile(`.+`)}}}
```

Comment:
rather than recompiling the regex every time, the compilation can be done in an `init()` function, then [`Regexp.Copy()`](https://golang.org/pkg/regexp/#Regexp.Copy) can be used here.